### ISSUE TITLE
Investigate whether search lambda (search_handler.py ) can be combined with api_handler.py

### ISSUE THREAD

**Issue by TejasRGitHub** [data-dot-all/dataall#1302]
**Is your idea related to a problem? Please describe.**
After this potential change -  https://github.com/data-dot-all/dataall/pull/1236/. The search_handler and the api_handler have a lot of common code. 

**Describe the solution you'd like**
Combine search_handler.py and api_handler.py without losing any performance. This will reduce the use of one lambda. Throttling should be checked as in this case both gql and the catalog search request would be coming on a single lambda 

*P.S. Don't attach files. Please, prefer add code snippets directly in the message body.*

**Comment by dlpzx** [comment]
Absolutely, thanks @TejasRGitHub for opening the issue. I think we shall consider it for v2.7 :)
